### PR TITLE
Display error message for NetworkErrorCode

### LINKED ISSUE #1654: Improve log message for connection errors
Logging a number is confusing:
`W0103 20:25:17.030653 31955 stmgr-client.cpp:94] Could not connect to stmgr stmgr-147 running at <host>:<port> due to: 1`

### PR DESCRIPTION
**Pull request by congwang**
This is to resolve #1654.

### CODE REVIEW

File: heron/common/src/cpp/network/network_error.h [hunk 1]
```diff
@@ -30,4 +33,12 @@ enum NetworkErrorCode {
   TIMEOUT
 };
 
+static inline std::ostream& operator<<(std::ostream& os, const NetworkErrorCode code) {
+  std::string strs[] = { "OK", "Connection error", "Read error", "Write error",
```
Comment (billonahill) [inline heron/common/src/cpp/network/network_error.h]: Can these instead be bound directly to the enum values themselves, instead of managed separately?
http://en.cppreference.com/w/cpp/language/enum

Comment (congwang) [inline heron/common/src/cpp/network/network_error.h]: I am not aware of any C++ syntax for this, we may archive it by a wrapper class or something like that, but I doubt it is worth any effort to improve it. Does it hurt readability to you?

Comment (objmagic) [inline heron/common/src/cpp/network/network_error.h]: this looks useful https://aantron.github.io/better-enums/tutorial/Conversions.html#Strings and easy to include in, if we bother to use it at all...

Comment (congwang) [inline heron/common/src/cpp/network/network_error.h]: There is always some way to do it. Again, why do you think it is worth our effort to do it? For me, we only have one case so far, no need to bother a complex way to make it unnecessarily elegant. If you believe it is worth, you are welcome to do it in a separate PR.

Comment (objmagic) [inline heron/common/src/cpp/network/network_error.h]: just pointing out. i have no opinion since C++ language does not have the capability of allowing people to use enum reflection easily.

**Comment by objmagic** [comment]
can you check out the CI?

**Comment by congwang** [comment]
@objmagic IntegrationTest_MultiSpoutsMultiTasks fails.

**Comment by congwang** [comment]
There is a new CI failure:

```
[ RUN      ] StMgr.test_tmaster_restart_on_new_address
src/central_freelist.cc:333] tcmalloc: allocation failed 24576 
Unknown exception caught in global handler
```

**Comment by objmagic** [comment]
Thanks. I'll just restart the CI.

**Comment by objmagic** [comment]
The PR CI is fine. Feel free to merge it.

**Comment by objmagic** [comment]
The PR CI check passed. Merged.
